### DISCUSSION TITLE
Would it make sense to use Mold for embedded?

### DISCUSSION THREAD

**Discussion by unref-ptr** [rui314/mold#1261] [Q&A]
I stumbled upon the project and wanted to know if for embedded targets MCUs it would make sense?

I tried using it with the Zephyr RTOS project but I got the error 

mold: fatal: zephyr/linker_zephyr_pre0.cmd:3: MEMORY
                                       ^ unknown linker script token
                                       
See https://sourceware.org/binutils/docs/ld/MEMORY.html

**Comment by rui314** [comment]
It does, but we'll do that without the linker script.

**Reply by unref-ptr** [reply]
Would it be possible to support the `MEMORY` section? Linker scripts are typically used in the linkage process for MCUs to allocate resources (interrupt vector tables, flash data, sram,etc) in particular addresses that are hardware dependent.

I see the parsing is done [here](https://github.com/rui314/mold/blob/a25226eb13abda6065ceaf94af6391b314a245c5/elf/linker-script.cc#L230) but I am not sure if this would be too much effort or difficult to try to implement.

**Reply by unref-ptr** [reply]
I just read the issue https://github.com/rui314/mold/issues/243. This gave me some context on why you dont fully support the linker script.

As far as I know all toolchains, and frameworks for embedded rely on linker scripts.  So perhaps at this point in time mold is not intended for these targets. It would be helpful if this was mentioned in the Readme.md 👍

**Comment by tannewt** [comment]
@rui314 I emailed (from my adafruit.com email) you a while ago to collaborate on embedded support. I also don't like linker scripts and would like something better. I work on CircuitPython which would benefit from this.
